xmodule/services.py

```diff
@@ -223,8 +222,7 @@ def rebind_noauth_module_to_user(self, block, real_user):
 
         block.scope_ids = block.scope_ids._replace(user_id=real_user.id)
         # now bind the module to the new ModuleSystem instance and vice-versa
-        block.runtime = inner_system
-        inner_system.xmodule_instance = block
+        block.runtime.xmodule_instance = block
```

Comment:
Thanks for pointing this out. I have removed this line, since this is redundant.

In `ModuleSystem`, `xmodule_instance` was used [here](https://github.com/openedx/edx-platform/blob/197fef9c2b31de27a5a462bcdd2d8e7f167a492c/xmodule/x_module.py#L1703-L1709) for `ajax_url` property.

While merging `ModuleSystem` and `DescriptorSystem`, I had removed this `ajax_url` property, since this was never referenced anywhere in the `edx-platform`. Instead, the `ajax_url` property was referenced from the [XModuleToXBlockMixin](https://github.com/openedx/edx-platform/blob/197fef9c2b31de27a5a462bcdd2d8e7f167a492c/xmodule/x_module.py#L764-L773) class, which is mixed-in to most XBlock classes.

However, this reference to `xmodule_instance` got me thinking if the `ajax_url` property was indeed referenced from somewhere outside `edx-platform`, and if I should add this property back to `DescriptorSystem`.